kvcache/causal.go

```diff
@@ -230,7 +243,7 @@ func (c *Causal) buildMask(ctx ml.Context, positions []int32, seqs []int) (ml.Te
 	}
 
 	if c.config.MaskDType != ml.DTypeF32 {
-		out := ctx.Empty(c.config.MaskDType, maskTensor.Shape()...)
+		out := ctx.Input().Empty(c.config.MaskDType, maskTensor.Shape()...)
```

Comment:
This case only triggers for flash attention. For non-flash attention the mask won't be copied and so won't be marked as an input.

Reply:
I've added the missing method